prusti-specs/src/specifications/preparser.rs

```diff
@@ -289,557 +58,547 @@ impl Parse for SpecEntArgs {
+        let mut t = vec![];
+        while !self.peek_operator(delimiter) && !self.tokens.is_empty() {
+            t.push(self.pop().unwrap());
+        }
+        let mut stream = TokenStream::new();
+        stream.extend(t.into_iter());
+        stream
     }
-    fn error_expected_bracket(&self) -> syn::Error {
-        syn::Error::new(self.input.span, "expected `[`")
+    /// pop tokens into a new stream for syn2 until the given operator character
+    fn create_stream_remaining(&mut self) -> TokenStream {
+        let mut t = vec![];
+        while !self.tokens.is_empty() {
+            t.push(self.pop().unwrap());
+        }
+        let mut stream = TokenStream::new();
+        stream.extend(t.into_iter());
+        stream
     }
-    fn error_expected_comma(&self) -> syn::Error {
-        syn::Error::new(self.input.span, "expected `,`")
+    /// get the span of the blamed token
+    fn get_error_span(&self) -> Span {
+        if let Some(span) = self.last_span {
+            span
+        } else if let Some(token) = self.tokens.front() {
+            token.span()
+        } else if let Some(span) = self.source_span {
+            span
+        } else {
+            Span::call_site()
+        }
     }
-    fn error_expected_or(&self) -> syn::Error {
-        syn::Error::new(self.input.span, "expected `|`")
+    /// complain about expecting a token
+    fn error_expected(&self, what: &str) -> syn::Error {
+        syn::Error::new(self.get_error_span(), format!("expected {}", what))
     }
-    fn error_expected_triggers(&self) -> syn::Error {
-        syn::Error::new(self.input.span, "expected `triggers`")
+    fn error_no_quantifier_arguments(&self) -> syn::Error {
+        syn::Error::new(self.get_error_span(), "a quantifier must have at least one argument")
     }
-    fn error_expected_specification(&self) -> syn::Error {
-        syn::Error::new(self.input.span, "expected `requires` or `ensures`")
+    fn error_expected_tuple(&self, span: Span) -> syn::Error {
+        syn::Error::new(span, "`triggers` must be an array of tuples containing Rust expressions")
     }
-    fn error_expected_equals(&self) -> syn::Error {
-        syn::Error::new(self.input.span, "expected `=`")
+    fn error_unexpected(&self) -> syn::Error {
+        syn::Error::new(self.get_error_span(), "unexpected token")
     }
-    fn error_expected_tuple(&self) -> syn::Error {
-        syn::Error::new(self.input.span, "`triggers` must be an array of tuples containing Rust expressions")
+    fn error_no_implies(&self, span: Span) -> syn::Error {
+        syn::Error::new(span, "`==>` cannot be part of Rust expression")
     }
-    fn error_ambiguous_expression(&self) -> syn::Error {
+    fn error_ambiguous_expression(&self, span: Span) -> syn::Error {
         syn::Error::new(
-            self.input.span,
+            span,
             "found `||` and `&&` in the same subexpression. \
             Hint: add parentheses to clarify the evaluation order.")
     }
```

Comment:
If some of these `error_*` functions are used only once, I don't see much value in them. Just inline the error message.

Reply:
I think keeping the `error_*` functions together keeps the error message strings together as well.